src/baldr/incident_singleton.h

```diff
@@ -0,0 +1,301 @@
+#pragma once
+
+#include "baldr/graphreader.h"
+#include "midgard/sequence.h"
+
+#include <chrono>
+#include <condition_variable>
+#include <ctime>
+#include <memory>
+#include <mutex>
+#include <thread>
+
+#include <google/protobuf/io/coded_stream.h>
+#include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+
+namespace {
+
+#ifdef _MSC_VER
+#define MTIME(st_stat) st_stat.st_mtime
+#elif __APPLE__
+#define MTIME(st_stat) st_stat.st_mtime
+#else
+#define MTIME(st_stat) st_stat.st_mtim.tv_sec
+#endif
+
+constexpr time_t DEFAULT_MAX_LOADING_LATENCY = 60;
+constexpr size_t DEFAULT_MAX_LATENT_COUNT = 5;
+
+/**
+ * Read the contents of a file into an incident tile
+ * @param filename   name of the file on the file system to read into memory
+ * @return a shared pointer with the data of the tile or an empty pointer if it could not be read
+ */
+std::shared_ptr<const valhalla::IncidentsTile> read_tile(const std::string& filename) {
+  // crack the file open
+  std::ifstream file(filename, std::ios::in | std::ios::binary);
+  if (!file.is_open()) {
+    LOG_WARN("Incident Watcher failed to open " + filename);
+    return {};
+  }
+
+  // prepare a stream for parsing
+  std::string buffer((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+  google::protobuf::io::ArrayInputStream as(static_cast<const void*>(buffer.c_str()), buffer.size());
+  google::protobuf::io::CodedInputStream cs(
+      static_cast<google::protobuf::io::ZeroCopyInputStream*>(&as));
```

Comment:
This could potentially be collapsed to
```
std::ifstream fin(filename);
ASSERT_TRUE(fin.is_open());

valhalla::incidents::IncidentsTile tile;
tile.ParseFromIstream(&fin);
```

Reply:
sadly not available in our version of protobuf

Reply:
Ignore above, our version of protobuf does not support this.